quakeworx/earthquake-tsunami/ttpv_qwx.ipynb

```diff
@@ -0,0 +1,539 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0d55b0d4-70a4-4bbb-b18e-6613235b9fb4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install seissolxdmf\n",
+    "import sys\n",
+    "sys.path.append(\"/home/qwxdev/.local/lib/python3.12/site-packages\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "id": "77772096-e8cc-4725-945b-229e4b6dbe25",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from matplotlib import pyplot as plt\n",
+    "\n",
+    "import seissolxdmf as seisx\n",
+    "from mpl_toolkits.mplot3d.art3d import Poly3DCollection\n",
+    "import matplotlib.colors as mcolors"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9e97870c-97af-4312-99a7-6dc7d23d36f3",
+   "metadata": {},
+   "source": [
+    "**Please use your job name here.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "id": "f47136c0-1d79-43fb-8145-0828dd4df64c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Asssuming you job name was \"SeisSol_TTPV1\"\n",
+    "job_name = \"SeisSol_TTPV1\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 50,
+   "id": "a8e7ab64-eb81-4d4b-98ba-b7b38c9e819c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "benchmark = \"ttpv1\" # we start with \"ttpv1\"; for those who run the second benchmark use \"ttpv2\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8c860e3a-125e-4469-b5a2-fea17e5b2787",
+   "metadata": {},
+   "source": [
+    "#### Fault output\n",
+    "\n",
+    "We first visualize what we call the **[fault output](https://seissol.readthedocs.io/en/latest/fault-output.html)** generated by SeisSol. We can visualize all processes occurring across the fault surface during dynamic rupture, e.g. slip rates, slip, shear and normal stresses, effective friction, etc.\n",
+    "\n",
+    "Feel free to explore all of these field, which are documented [here](https://seissol.readthedocs.io/en/latest/fault-output.html#outputmask). For now, we focus on the accumulated fault slip (\"ASl\"). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 51,
+   "id": "49442394-251f-4889-862b-65fdf2c15694",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load data\n",
+    "xdmfFilename = \"../../../{}/outputs/{}-fault.xdmf\".format(job_name,benchmark)\n",
+    "# If run locally, give the absolut path\n",
+    "#xdmfFilename = \"/Users/fkutschera/Documents/PhD/Quakeworx/simulations/SeisSol_ttpv1_branch_test_1/outputs/ttpv1-fault.xdmf\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 88,
+   "id": "1492cb89-67d9-47bb-ae25-952721e49252",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_fault(xdmfFilename, ndt=None, var=\"ASl\", vmin=None, vmax=None, cmap=\"magma_r\"):\n",
```

Comment:
in the end plot_fault and plot_surface could be moved to an external python file?
(as I don't think it is important for the user to read what is inside).